apps/api/src/database/factories/movieTitle.ts

```diff
@@ -1,103 +1,52 @@
-import { MovieTitle } from '@prisma/client'
+import { Movie, MovieTitle } from '@prisma/client'
 import { Factory } from 'fishery'
-import { faker } from '@faker-js/faker'
+import { allFakers, faker } from '@faker-js/faker'
 
-import { supportedLanguageLocaleCodes, Language } from '../../constants'
 import prisma from '../client'
+import { supportedLanguages } from '../../constants'
 
-class MovieTitleFactory extends Factory<MovieTitle> {
+import { movieFactory } from './movie'
 
+interface IMovieTitleFactory extends MovieTitle {
+  movie?: Movie
 }
+class MovieTitleFactory extends Factory<IMovieTitleFactory> {}
 
 export const movieTitleFactory = MovieTitleFactory.define(({ sequence, onCreate, params }) => {
   onCreate(async (movieTitle) => {
-    // Require movieId if creating a movieTitle in the DB
-    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-    if (!params.movieId) {
-      throw new Error('[movieTitleFactory] movieId must be provided')
+    // Create a movie if not provided
+    if (!movieTitle.movie) {
+      movieTitle.movie = await movieFactory.create(movieTitle.movie)
     }
 
     return await prisma.movieTitle.create({
-      data: movieTitle
+      data: {
+        title: movieTitle.title,
+        language: movieTitle.language,
+        image: movieTitle.image,
+        movie: {
+          connect: {
+            id: movieTitle.movie.id
+          }
+        }
+      },
+      include: {
+        movie: true
+      }
     })
   })
 
-  const language = (params?.language !== undefined && params.language in supportedLanguageLocaleCodes)
-    ? params.language as Language
-    : faker.helpers.arrayElement(supportedLanguageLocaleCodes)
-
-  const title = params.title ?? generateMovieTitles([language])[language]
+  const language = supportedLanguages.find(supportedLanguage => supportedLanguage === params.language) ?? 'en'
+  // @ts-expect-error - TS doesn't know that allFakers[language] is a valid faker instance
+  const title = params.title ?? allFakers[language].word.words({ count: { min: 3, max: 10 } })
 
   return {
-    movieTitleId: sequence,
-    movieId: params.movieId ?? sequence,
```

Comment:
you can leave `params` aside as it will already automatically replace with whatever data you put in `.build()` method